app/index.js

```diff
@@ -55,62 +77,71 @@ var AspnetGenerator = yeoman.generators.Base.extend({
             name: 'Unit test project',
             value: 'unittest'
           }
-      ]
-    }];
+        ]
+      }];
 
-    this.prompt(prompts, function(props) {
-      this.type = props.type;
+      this.prompt(prompts, function (props) {
+        this.type = props.type;
 
-      done();
-    }.bind(this));
+        done();
+      }.bind(this));
+    }
   },
 
-  askForName: function() {
-    var done = this.async();
-    var app = '';
-    switch (this.type) {
-      case 'empty':
-        app = 'EmptyApplication';
-        break;
-      case 'console':
-        app = 'ConsoleApplication';
-        break;
-      case 'web':
-        app = 'WebApplication';
-        break;
-      case 'webbasic':
-        app = 'WebApplicationBasic';
-        break;
-      case 'webapi':
-        app = 'WebAPIApplication';
-        break;
-      case 'nancy':
-        app = 'NancyApplication';
-        break;
-      case 'classlib':
-        app = 'ClassLibrary';
-        break;
-      case 'unittest':
-        app = 'UnitTest';
-        break;
+  buildTemplateData: function () {
```

Comment:
This should be changed I think. As per Yeoman docs helpers/private methods needs - in this implementation - underscore. See pt. 1:
http://yeoman.io/authoring/running-context.html.
Otherwise `buildTemplateData` will be called by generator itself